### ISSUE TITLE
Pandas ReadCSV assumptions

### ISSUE THREAD

**Issue by subdavis** [girder/metabulo#30]
https://github.com/girder/metabulo/blob/6735897c92ce206dea1bc9bfc7ea608e501579fc/metabulo/models.py#L137-L162

These assumptions are probably fine in most cases, but might occasionally cause trouble.

I'm more concerned about mapping between row/col indexes and pandas indexes: `index + 1` or `index - 1` is going to end up elsewhere in code and will read poorly.

## Solution

We will eventually have to scan the input CSV and pick out what we think the primary and secondary keys are, but I don't think it's necessary that the pandas data frame have header or index_col set.

```python
    pandas.read_csv(uri, header=None)
```

Then, to get the data table after keys, attributes, and masked rows are removed:

```python
    column_data = filter(lambda col: col.column_type == 'numeric', columns)
    column_data_indexes = list(map(lambda col: col.column_index, column_data))

    row_data = filter(lambda row: row.row_type == 'sample', rows)
    row_data_indexes = list(map(lambda row: row.row_index, row_data))
    
    filtered = frame.iloc[row_data_indexes, column_data_indexes] ## indexes match up without off-by-one issues.
```

## Priority

This is low priority.  Just wanted to log my thoughts so I don't forget what's going on.

**Comment by jbeezley** [comment]
I'm working on some changes similar to what you suggested.  As of now, the table provided through the API doesn't reflect any changes made to the row/column models.  I think this is the source of your concern.

FWIW,  I'm planning on keeping the following assumptions until we have a reason not to:
* All csv files have a header at row 0
* The primary key (if it exists) is in the first column (otherwise the default `RangeIndex` is used)

**Comment by subdavis** [comment]
> keeping the following assumptions until we have a reason not to: header at row 0

The best reason I can think of not to do that is the user expects to be able to change the primary key.  If I try to make the primary key row[1] through the UI, row 0 should be demoted either to regular data or to secondary attribute / secondary key.

As a user, I have a hunch that locking the top row would occasionally be quite frustrating.

> As of now, the table provided through the API doesn't reflect any changes made to the row/column models.

As long as there remains a way of getting the raw uploaded table without consideration of the changes made to row/col models, any changes shouldn't impact existing UI much.

**Comment by subdavis** [comment]
Actually, as I recall, Tom showed us a data set where row 0 was blank (it had some comments in cell (0,0) and 1 had the primary keys last week.

**Comment by jbeezley** [comment]
Fixed by #49
